ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/PendingPeerRequest.java

```diff
@@ -80,8 +83,8 @@ private synchronized void sendRequest(final EthPeer peer) {
   }
 
   private Optional<EthPeer> getPeerToUse() {
-    // return the assigned peer if still valid, otherwise switch to another peer
-    return peer.filter(p -> !p.isDisconnected()).isPresent()
+    // If a peer was supplied, use it, otherwise grab one from EthPeers
+    return peer.isPresent()
```

Comment:
This could break the behaviour of many of the tasks that aren't using retry switching tasks. Is their a plan to change those tasks to use retry switching tasks instead?

Reply:
It's difficult to figure out which ones would benefit from being moved due to the complexity of the code. I'm starting to think maybe we should close this PR and focus on a larger refactor/rewrite to simplify as I've described in other comments.